### ISSUE TITLE
Suggestions for improving dev-set performance.

### ISSUE THREAD

**Issue by Feynman27** [baidu-research/ba-dls-deepspeech#4]
(I apologize if this question is better suited for StackOverflow, but I figure posting it here will reach the right audience in a shorter amount of time.) 

I'm training this CTC-cost model on the [Librispeech](http://www.openslr.org/12/) "train-other-500" dataset, which contains 500 hours of speech audio+transcripts. I'm using the "dev-other" data set for development, which is apparently a more challenging audio set to model. 

I trained the model over 20 epochs and have provided the distribution of the costs below. 
![image](https://cloud.githubusercontent.com/assets/12432112/19937476/d38f0184-a0f7-11e6-95a2-c8e5e4b190c4.png)

The weights are updated according to [Nesterov momentum](http://cs231n.github.io/neural-networks-3/#sgd). 

Since the validation performance plateaus at around iter=25000, I decided to checkpoint the model here and continue running the model using an exponential learning-rate decay schedule. The learning rate is decreased after each epoch (starting from iter=25000). The CTC costs using this learning-rate decay schedule are shown below after a few epochs:

![image](https://cloud.githubusercontent.com/assets/12432112/19937705/b86776ba-a0f8-11e6-95f3-a0db0063dd77.png)

Unfortunately, this strategy doesn't appear to improve the model performance. Does anyone have any suggestions on how to improve the model other than what I've described above?

**Comment by srvinay** [comment]
From the looks of it, your model seems to have high variance. You should try reducing the initial learning rate, add in regularization (dropout/augment with noise) or play with the model architecture if these ideas don't work.

**Comment by dylanbfox** [comment]
You can also try increasing the data you're training on. By default the max wav length is set to 10 seconds (https://github.com/baidu-research/ba-dls-deepspeech/blob/master/data_generator.py#L53-L54) which excludes a good portion of the data in the LibriSpeech corpora. Longer utterances most likely will require more memory usage though.
